nixops/backends/__init__.py

```diff
@@ -427,10 +452,21 @@ def _get_scp_name(self) -> str:
         return ssh_name
 
     def _fmt_rsync_command(self, *args: str, recursive: bool = False) -> List[str]:
-        master = self.ssh.get_master()
+        master = self.ssh.get_master(user=self.ssh_user)
 
         ssh_cmdline: List[str] = ["ssh"] + self.get_ssh_flags() + master.opts
         cmdline = ["rsync", "-e", nixops.util.shlex_join(ssh_cmdline)]
+
+        if self.ssh_user != "root":
+            cmdline.extend(
+                [
+                    "--rsync-path",
+                    nixops.util.shlex_join(
+                        self.ssh.privilege_escalation_command + ["rsync"]
```

Comment:
I wonder if we should make  a note in the docs for priv esc cmd about ending their command with `--`? We can't universally add one here, but they should probably take care to tell their tool to "okay, stop parsing  your args now".